### ISSUE TITLE
sharness test command coverage

### ISSUE THREAD

**Issue by whyrusleeping** [jbenet/go-ipfs#906]
Making this issue to track all CLI commands that need to be tested and conditions

Module |Online Test| Offline Test|
-----------|----------------|-----------------
object   |                     |          
ls           |                    |                      
cat        |                     |

**Comment by chriscool** [comment]
Ok, I think the table above is roughly uptodate now.

**Comment by whyrusleeping** [comment]
@chriscool, thanks a bunch for that!

**Comment by cryptix** [comment]
Maybe we should commit this table to the `tests` folder?

**Comment by whyrusleeping** [comment]
Might be a good idea. make it the `readme.md` inside tests

**Comment by jbenet** [comment]
:+1: for readme.

I think these table (or tables) should also include what flags + arg combinations are tested.

**Comment by chriscool** [comment]
If this issue is about adding all the missing sharness tests, maybe it could be broken down into many issues or items.

If it was just about adding the current coverage into a file then it is done since PR #954 has been merged.

**Comment by whyrusleeping** [comment]
@chriscool Agreed on breaking this down into many issues. We can probably close this issue as well, following input from @jbenet

**Comment by jbenet** [comment]
Agreed on breaking it down.




I want to use this issue to track others. We can have a list with checkboxes with each other issue that's needed. (Part of it is how it plays with github milestones and so on)



—

**Comment by Stebalien** [comment]
Updated:

| Module    | Online Test             | Offline Test |
| ---       | ---                     | ---          |
| object    | t0051                   | t0051        |
| ls        | t0045                   | t0045        |
| cat       | t0040                   | t0040        |
| dht       | t0170                   | N/A          |
| bitswap   | t0220                   | N/A          |
| block     |                         | t0050        |
| daemon    | t0030                   | N/A          |
| init      | N/A                     | t0020        |
| add       | t0040                   | t0040        |
| config    | t0021                   | t0021        |
| version   |                         | t0010        |
| ping      |                         | N/A          |
| diag      | t0065, (missing: t0151) | t0151        |
| mount     | t0030                   | N/A          |
| name      | t0110                   | t0100        |
| pin       | t0080,t0081,t0085       |              |
| get       | t0090                   |              |
| refs      | t0080                   |              |
| repo gc   | t0080                   |              |
| id        |                         | t0010        |
| bootstrap | t0120                   | t0120        |
| swarm     | t0140                   | N/A          |
| update    |                         |              |
| commands  |                         | t0010        |

Critically missing:

* [ ] `ipfs ping`

Most of the missing tests are just cases where we test online or offline but not
both.
